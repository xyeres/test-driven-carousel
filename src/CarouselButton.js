import React from 'react';
import PropTypes from 'prop-types';
import style from 'styled-components'


const Button = style.button`
  padding: 0.5rem;
  border:none;
  background-color: lightblue;
  margin-right:1rem;
  color: ${(props) => 
    props.color ? props.color : 'darkpink'  
  }
`

const CarouselButton = ({color, ...props}) => <Button color={color} {...props} />;

CarouselButton.propTypes = {
  color: PropTypes.string,
  Button: PropTypes.elementType,
  children: PropTypes.node.isRequired,
};

CarouselButton.defaultProps = {
  Button: Button,
};


export default CarouselButton;
