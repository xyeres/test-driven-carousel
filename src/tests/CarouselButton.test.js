import React from 'react'; // 1
import { mount } from 'enzyme'; // 2
import CarouselButton from '../CarouselButton';
import styled from 'styled-components';

describe('CarouselButton', () => {
  const text = 'Button text';
  let mounted;

  beforeEach(() => {
    mounted = mount(<CarouselButton>{text}</CarouselButton>);
  });

  it('renders correctly', () => {
    expect(mounted.find('button')).toMatchSnapshot();
  });

  it('passes `children` through to the <button>', () => {
    expect(mounted.prop('children')).toBe(text);
  });

  it('allows color prop to be overriden', () => {
    const color = 'lightpurple';
    mounted.setProps({ color });
    expect(mounted.prop('color')).toBe(color);
  });

  it('passes other props through to the <button>', () => {
    const onClick = () => {};
    const className = 'my-carousel-button';
    const dataAction = 'prev';
    mounted.setProps({ onClick, className, 'data-action': dataAction }); // 1
    expect(mounted.prop('onClick')).toBe(onClick);
    expect(mounted.prop('className')).toBe(className);
    expect(mounted.prop('data-action')).toBe(dataAction);
  });
});
