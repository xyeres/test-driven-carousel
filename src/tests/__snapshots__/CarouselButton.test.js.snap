// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselButton renders correctly 1`] = `
.c0 {
  padding: 0.5rem;
  border: none;
  background-color: lightblue;
  margin-right: 1rem;
  color: darkpink;
}

<button
  className="c0"
>
  Button text
</button>
`;
